### PR TITLE
Tidy up GiftedChat component

### LINKED ISSUE #2370: Excessive rerenders because of context changes
Today I used [patch-package](https://github.com/ds300/patch-package) to patch `react-native-gifted-chat@2.0.1` for the project I'm working on.

Because GiftedChat uses an object as the context value it causes every component that listens to context to rerender whenever GiftedChat rerenders.  This causes all the message bubbles, message texts and many other components to rerender.  By saving the context values to state it prevents these context changes from happening. This probably helps some of the performance issues that #2366, #2335, #2336 are facing.

Using the example on the old react documentation for the workaround.
https://legacy.reactjs.org/docs/context.html#caveats

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/react-native-gifted-chat/lib/GiftedChat.js b/node_modules/react-native-gifted-chat/lib/GiftedChat.js
index a91d004..4d03071 100644
--- a/node_modules/react-native-gifted-chat/lib/GiftedChat.js
+++ b/node_modules/react-native-gifted-chat/lib/GiftedChat.js
@@ -44,6 +44,7 @@ class GiftedChat extends React.Component {
             typingDisabled: false,
             text: undefined,
             messages: undefined,
+            contextValues:{},
         };
         this.getLocale = () => this._locale;
         this.onKeyboardWillShow = (e) => {
@@ -193,6 +194,11 @@ class GiftedChat extends React.Component {
         this.initLocale();
         this.setMessages(messages || []);
         this.setTextFromProp(text);
+        this.setState({contextValues: {
+            actionSheet: this.props.actionSheet ||
+                (() => { var _a; return (_a = this._actionSheetRef.current) === null || _a === void 0 ? void 0 : _a.getContext(); }),
+                getLocale: this.getLocale,
+        }})
     }
     componentWillUnmount() {
         this.setIsMounted(false);
@@ -416,13 +422,7 @@ class GiftedChat extends React.Component {
     }
     render() {
         if (this.state.isInitialized === true) {
-            const actionSheet = this.props.actionSheet ||
-                (() => { var _a; return (_a = this._actionSheetRef.current) === null || _a === void 0 ? void 0 : _a.getContext(); });
-            const { getLocale } = this;
-            return (<GiftedChatContext.Provider value={{
-                    actionSheet,
-                    getLocale,
-                }}>
+            return (<GiftedChatContext.Provider value={this.state.contextValues}>
           <View testID={TEST_ID.WRAPPER} style={styles.wrapper}>
             <ActionSheetProvider ref={this._actionSheetRef}>
               <View style={styles.container} onLayout={this.onMainViewLayout}>
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

### PR DESCRIPTION
**Pull request by amerikan**
# Purpose

- adds Ref suffix to useRef variables for readability 
- consolidates state setter when mounted into a single setter
- removes superfluous helper functions that are no longer needed

~~Unused props are not removed in this PR. Not sure why they're there, but thought I just leave them alone for now.~~ Removed unused props.

### CODE REVIEW

File: src/GiftedChat.tsx [hunk 1]
```diff
@@ -447,7 +376,7 @@ function GiftedChat(props: GiftedChatProps) {
   const getBasicMessagesContainerHeight = (
     composerHeight = state.composerHeight,
   ) => {
-    return getMaxHeight()! - calculateInputToolbarHeight(composerHeight!)
+    return maxHeightRef.current! - calculateInputToolbarHeight(composerHeight!)
```
Comment (Johan-dutoit) [inline src/GiftedChat.tsx:379]: potential for `NaN`, as max height could be undefined.

**Comment by Johan-dutoit** [comment]
Otherwise LGTM

**Comment by amerikan** [comment]
@Johan-dutoit fixed the type errors, but don't think the Github actions is set up correctly.

**Comment by Johan-dutoit** [comment]
> @Johan-dutoit fixed the type errors, but don't think the Github actions is set up correctly.

@amerikan my bad, was "working" on the PR.  Fixed now and also added node 18.

**Review by Johan-dutoit — APPROVED** [review]

**Pull request merged**